types/hapi/definitions/request/request.d.ts

```diff
@@ -0,0 +1,216 @@
+     */
+    plugins: PluginsStates;
+
+    /**
+     * An object where each key is the name assigned by a route pre-handler methods function. The values are the raw values provided to the continuation function as argument. For the wrapped response object, use responses.
+     */
+    readonly pre: Util.Dictionary<object>;
+
+    /**
+     * Access: read / write (see limitations below).
+     * The response object when set. The object can be modified but must not be assigned another object. To replace the response with another from within an extension point, use reply(response) to override with a different response. Contains null when no response has been set (e.g. when a request terminates prematurely when the client disconnects).
+     */
+    response: ResponseObject | null;
+
+    /**
+     * Same as pre but represented as the response object created by the pre method.
+     */
+    readonly preResponses: Util.Dictionary<object>;
+
+    /**
+     * By default the object outputted from node's URL parse() method. Might also be set indirectly via request.setUrl in which case it may be a string (if url is set to an object with the query attribute as an unparsed string).
+     */
+    readonly query: any;
+
+    /**
+     * An object containing the Node HTTP server objects. Direct interaction with these raw objects is not recommended.
+     * * req - the node request object.
+     * * res - the node response object.
+     */
+    readonly raw: {
+        req: http.IncomingMessage;
+        res: http.ServerResponse;
+    };
+
+    /**
+     * The request route information object and method
+     * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-requestroute)
+     * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-requestrouteauthaccessrequest)
+     */
+    readonly route: RequestRoute;
+
+    /**
+     * Access: read only and the public server interface.
+     * The server object.
+     */
+    server: Server;
+
+    /**
+     * An object containing parsed HTTP state information (cookies) where each key is the cookie name and value is the matching cookie content after processing using any registered cookie definition.
+     */
+    readonly state: Util.Dictionary<any>;
+
+    /**
+     * The parsed request URI.
+     */
+    readonly url: url.Url;
+
+    /**
+     * Returns a response which you can pass into the reply interface where:
+     * @param source - the value to set as the source of the reply interface, optional.
+     * @param options - options for the method, optional.
+     * @return ResponseObject
+     * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-requestgenerateresponsesource-options)
+     */
+    generateResponse(source: string | object | null, options?: {variety?: string; prepare?: Function; marshal?: Function; close?: Function; }): ResponseObject;
```

Comment:
Please enhance the typings here, the docs have a much better definition than just `Function`.